src/compas_viewer/scene/nurbssurfaceobject.py

```diff
@@ -17,52 +19,36 @@ class NurbsSurfaceObject(ViewerSceneObject, GeometryObject):
 
     def __init__(self, surface: NurbsSurface, **kwargs):
         super().__init__(geometry=surface, **kwargs)
+        self.geometry: NurbsSurface
 
         # LINEARDEFLECTION not implemented in NurbsSurface.
-        self.u = kwargs.get("u", int(16 + (0 * self.LINEARDEFLECTION)))
-        self.v = kwargs.get("v", int(16 + (0 * self.LINEARDEFLECTION)))
-
-        self._triangles = [list(point) for triangle in surface.to_triangles(nu=self.u, nv=self.v) for point in triangle]
-
-    def _read_points_data(self) -> DataType:
-        positions = [Point(*pt) for pt in flatten(self.geometry.points)]
-        colors = [self.pointscolor.get(i, self.pointscolor["_default"]) for i in range(len(positions))]  # type: ignore
-        elements = [[i] for i in range(len(positions))]
-        return positions, colors, elements
+        self.u = int(16 + (0 * self.LINEARDEFLECTION))
+        self.v = int(16 + (0 * self.LINEARDEFLECTION))
 
-    def _read_lines_data(self) -> DataType:
-        positions = [Point(*pt) for pt in flatten(self.geometry.points)]
-        colors = [self.linescolor.get(i, self.linescolor["_default"]) for i in range(len(positions))]  # type: ignore
-        count = 0
+    @property
+    def points(self) -> Optional[list[Point]]:
+        """The points to be shown in the viewer."""
+        points = []
+        for row in self.geometry.points:
```

Comment:
for a nurbssurface, these points are the control points...

Reply:
I  think we want to display the control points no?